src/main/java/se/diabol/jenkins/pipeline/domain/Pipeline.java

```diff
@@ -114,7 +119,61 @@ public void setChanges(List<Change> changes) {
         return changes;
     }
 
+    @Exported
+    public long getTotalBuildTime() {
+        return totalBuildTime;
+    }
 
+    public void setTotalBuildTime() {
+        if (stages.size() == 0) {
+            this.totalBuildTime = 0L;
+        } else {
+            List<Route> allRoutes = new ArrayList<Route>();
+            calculatePipelineRoutes(getStages().get(0).getTasks().get(0), null, allRoutes);
+            long maxTime = 0L;
+            for (Route route : allRoutes) {
+                long buildTime = route.getTotalBuildTime();
+                if (buildTime > maxTime) {
+                    maxTime = buildTime;
+                }
+            }
+            this.totalBuildTime = maxTime;
+        }
+    }
+
+    private Route createRouteAndCopyTasks(final Route route, Task task) {
+        Route currentRoute = new Route();
+        if (route != null) {
+            currentRoute.setTasks(newArrayList(route.getTasks()));
+        }
+        currentRoute.addTask(task);
+        return currentRoute;
+    }
+
+    public void calculatePipelineRoutes(Task task, final Route route, List<Route> allRoutes) {
```

Comment:
I wanted to test this method directly so I made it public. Do you suggest making it private?